# HG changeset patch
# User Rik <rik@octave.org>
# Date 1546556401 28800
#      Thu Jan 03 15:00:01 2019 -0800
# Branch stable
# Node ID 05ce565644bdbab09b75f5ef1f680c139e215d6c
# Parent  ccacd7162520115360cff117355a7513d8dc2b53
ov-bool.cc: Fix static analyzer detected issues (bug #55347).

* ov-bool.cc (load_ascii, load_hdf5): Use "0.0" instead of "0." for floating
point 0 for clarity.

diff --git a/libinterp/octave-value/ov-bool.cc b/libinterp/octave-value/ov-bool.cc
--- a/libinterp/octave-value/ov-bool.cc
+++ b/libinterp/octave-value/ov-bool.cc
@@ -189,17 +189,17 @@ octave_bool::save_ascii (std::ostream& o
   os << "\n";
 
   return true;
 }
 
 bool
 octave_bool::load_ascii (std::istream& is)
 {
-  scalar = (octave_read_value<double> (is) != 0.);
+  scalar = (octave_read_value<double> (is) != 0.0);
 
   if (! is)
     error ("load: failed to load scalar constant");
 
   return true;
 }
 
 bool
@@ -289,17 +289,17 @@ octave_bool::load_hdf5 (octave_hdf5_id l
   double dtmp;
   if (H5Dread (data_hid, H5T_NATIVE_DOUBLE, octave_H5S_ALL, octave_H5S_ALL,
                octave_H5P_DEFAULT, &dtmp) < 0)
     {
       H5Dclose (data_hid);
       return false;
     }
 
-  scalar = (dtmp != 0.);
+  scalar = (dtmp != 0.0);
 
   H5Dclose (data_hid);
 
 #else
   octave_unused_parameter (loc_id);
   octave_unused_parameter (name);
 
   warn_load ("hdf5");
